src/parser/classes/AdsEngagementPanelContent.ts

```diff
@@ -0,0 +1,13 @@
+import { YTNode } from '../helpers.js';
+import { type RawNode } from '../index.js';
+
+export default class AdsEngagementPanelContent extends YTNode {
+  static type = 'AdsEngagementPanelContent';
+
+  hack: boolean;
+
+  constructor(data: RawNode) {
+    super();
+    this.hack = data.hack;
+  }
```

Comment:
Is this node required? Maybe it should be added to the list of ignored nodes in the parser as it's related to ads.